pkg/action/update_action.go

```diff
@@ -37,6 +39,33 @@ func UpdateAction(cfg Config) error {
 		return fmt.Errorf("failed to add steps: %w", err)
 	}
 
+	konfluxVersions, err := prowgen.ServerlessOperatorKonfluxVersions(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed to get serverless operator versions: %w", err)
+	}
+
+	soConfig := &prowgen.Config{
+		Repositories: []prowgen.Repository{
+			{Org: "openshift-knative", Repo: "serverless-operator"},
+		},
+	}
+	for _, release := range sortedKeys(konfluxVersions) {
+		branch := konfluxVersions[release]
+		if branch == "main" {
+			continue
+		}
+		if soConfig.Config.Branches == nil {
+			soConfig.Config.Branches = map[string]prowgen.Branch{}
+		}
+		soConfig.Config.Branches[branch] = prowgen.Branch{
+			Konflux: &prowgen.Konflux{
+				Enabled: true,
+			},
+		}
+	}
+	_, s := updateAction(soConfig)
+	steps = append(steps, s...)
```

Comment:
For me it is not really clear, why we update the actions here for SO, when we do the same (starting at line 69 again, as we have the SO config now in the config folder)

Reply:
Ah. This is for the release-x.y branches (which are not in the SO config in config folder), correct?

Reply:
Exactly

Reply:
Added a comment to explain the why

Reply:
awesome. Thanks :+1: